WordPress/Classes/ViewRelated/NUX/SignupDomainSuggestionViewController.swift

```diff
@@ -0,0 +1,244 @@
+    var helpBadge: WPNUXHelpBadgeLabel!
+    var helpButton: UIButton!
+
+    open var siteName: String?
+
+    var service: DomainsService?
+    private var siteTitleSuggestions: [String] = []
+    private var searchSuggestions: [String] = []
+    private var isSearching: Bool = false
+
+    required init?(coder aDecoder: NSCoder) {
+        super.init(coder: aDecoder)
+    }
+
+    override func awakeFromNib() {
+        super.awakeFromNib()
+        tableView.register(UINib(nibName: "SiteCreationDomainSearchTableViewCell", bundle: nil), forCellReuseIdentifier: SiteCreationDomainSearchTableViewCell.cellIdentifier)
+        tableView.register(UINib(nibName: "SiteCreationDomainsActivityTableViewCell", bundle: nil), forCellReuseIdentifier: SiteCreationDomainsActivityTableViewCell.cellIdentifier)
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        view.backgroundColor = WPStyleGuide.greyLighten30()
+
+        let (helpButtonResult, helpBadgeResult) = addHelpButtonToNavController()
+        helpButton = helpButtonResult
+        helpBadge = helpBadgeResult
+    }
+
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+
+        // only procede with initial search if we don't have site title suggestions yet (hopefully only the first time)
+        guard siteTitleSuggestions.count < 1,
+            let nameToSearch = siteName else {
+            return
+        }
+
+        suggestDomains(for: nameToSearch) { [weak self] (suggestions) in
+            self?.siteTitleSuggestions = suggestions
+            self?.tableView.reloadSections(IndexSet(integersIn: Sections.searchField.rawValue...Sections.siteTitleSuggestions.rawValue), with: .automatic)
+        }
+    }
+
+    private func suggestDomains(for searchTerm: String, addSuggestions: @escaping (_: [String]) ->()) {
+        guard !isSearching else {
+            return
+        }
+
+        isSearching = true
+
+        let api = WordPressComRestApi(oAuthToken: "")
+        let service = DomainsService(managedObjectContext: ContextManager.sharedInstance().mainContext, remote: DomainsServiceRemote(wordPressComRestApi: api))
+        tableView.reloadSections(IndexSet(integer: Sections.searchSuggestions.rawValue), with: .top)
+        service.getDomainSuggestions(base: searchTerm, success: { [weak self] (suggestions) in
+            self?.isSearching = false
+            addSuggestions(suggestions)
+            self?.tableView.reloadSections(IndexSet(integer: Sections.searchSuggestions.rawValue), with: .automatic)
+        }) { [weak self] (error) in
+            self?.isSearching = false
+        }
+    }
+}
+
+// MARK: - LoginWithLogoAndHelpViewController methods
+
+extension SignupDomainSuggestionViewController: LoginWithLogoAndHelpViewController {
+    func handleHelpButtonTapped(_ sender: AnyObject) {
+        displaySupportViewController(sourceTag: .wpComLogin)
```

Comment:
The other steps in the Site Creation process have new a sourceTag specific to the view. Can we do this here? Plus, this isn't Login. :)